test/support/seeds.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Cadet.Test.Seeds do
+  @moduledoc """
+  This module contains functions that seed more complex setups into the DB for tests.
+  """
+  import Cadet.Factory
+
+  def assessments do
+    if Application.get_env(:cadet, :environment) == :test do
+      # User and Group
+      avenger = insert(:user, %{name: "avenger", role: :staff})
+      mentor = insert(:user, %{name: "mentor", role: :staff})
+      group = insert(:group, %{leader: avenger, mentor: mentor})
+      students = insert_list(5, :student, %{group: group})
+      admin = insert(:user, %{name: "admin", role: :admin})
+      Enum.each([avenger, mentor] ++ students, &insert(:nusnet_id, %{user: &1}))
+
+      assessments =
+        Enum.reduce(
+          Cadet.Assessments.AssessmentType.__enum_map__(),
+          %{},
+          fn type, acc -> Map.put(acc, type, build_assessment(type, students)) end
```

Comment:
Partials? Haha

Reply:
I think I'd prefer to stick to the lambda syntax here. `&Map.put(&2, &1, insert_assessments(&1, students))` is a lot less readable